src/common/CMakeLists.txt

```diff
@@ -1,44 +1,88 @@
 ecbuild_enable_fortran(REQUIRED MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/module)
 
-ecbuild_add_library( TARGET cloudsc-common-lib
-    SOURCES
-        module/parkind1.F90
-        module/yoecldp.F90
-        module/yomcst.F90
-        module/yoethf.F90
-        module/yoephli.F90
-        module/yoecld.F90
-        module/yoecldp.F90
-        module/yophnc.F90
-        module/yomncl.F90
-        module/yomphyder.F90
-        module/fcttre_mod.F90
-        module/fccld_mod.F90
-        module/routines.F90
-        module/abor1.F90
-        module/timer_mod.F90
-        module/mycpu.c
-        module/ec_pmon_mod.F90
-	module/expand_mod.F90
-	module/file_io_mod.F90
-	module/hdf5_file_mod.F90
-        module/validate_mod.F90
-        module/cloudsc2_array_state_mod.F90
-        module/cloudsc_mpi_mod.F90
+list(APPEND CLOUDSC2_COMMON_SOURCES
+  module/parkind1.F90
+  module/yoecldp.F90
+  module/yomcst.F90
+  module/yoethf.F90
+  module/yoephli.F90
+  module/yoecld.F90
+  module/yoecldp.F90
+  module/yophnc.F90
+  module/yomncl.F90
+  module/yomphyder.F90
+  module/fcttre_mod.F90
+  module/fccld_mod.F90
+  module/routines.F90
+  module/abor1.F90
+  module/timer_mod.F90
+  module/mycpu.c
+  module/ec_pmon_mod.F90
+  module/expand_mod.F90
+  module/file_io_mod.F90
+  module/hdf5_file_mod.F90
+  module/validate_mod.F90
+  module/cloudsc2_array_state_mod.F90
+  module/cloudsc_mpi_mod.F90
+)
+
+
+ecbuild_add_library( TARGET cloudsc2-common-lib
+    SOURCES ${CLOUDSC2_COMMON_SOURCES}
     DEFINITIONS ${CLOUDSC_DEFINITIONS}
 )
-target_include_directories( cloudsc-common-lib PUBLIC $<INSTALL_INTERFACE:include> )
-target_include_directories( cloudsc-common-lib PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
-target_include_directories( cloudsc-common-lib PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/module>)
+target_include_directories( cloudsc2-common-lib PUBLIC $<INSTALL_INTERFACE:include> )
```

Comment:
Could we go to the same generator-expression-based inline definition as we do in CLOUDSC? As in add this to the `ecbuild_add_library()` call above? 
```
    PRIVATE_INCLUDES
        $<${HAVE_HDF5}:${HDF5_Fortran_INCLUDE_DIRS}>
    PUBLIC_INCLUDES
        $<INSTALL_INTERFACE:include>
        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/module>
```